bluebutton-data-pipeline-rif-load/src/main/java/gov/hhs/cms/bluebutton/data/pipeline/rif/load/RifLoader.java

```diff
@@ -444,12 +444,19 @@ private void processAsync(List<RifRecordEvent<?>> recordsBatch, PostgreSqlCopyIn
 						if (recordInDb == null) {
 							loadAction = LoadAction.INSERTED;
 							entityManager.persist(record);
+							Object recordInDbAfterUpdate = entityManager.find(record.getClass(), recordId);
 						} else {
 							loadAction = LoadAction.DID_NOTHING;
 						}
-					} else if (strategy == LoadStrategy.INSERT_NON_IDEMPOTENT) {
-						loadAction = LoadAction.INSERTED;
-						entityManager.persist(record);
+					} else if (strategy == LoadStrategy.INSERT_UPDATE_NON_IDEMPOTENT) {
+						if (rifRecordEvent.getRecordAction().equals(RecordAction.INSERT)) {
+							loadAction = LoadAction.INSERTED;
+							entityManager.persist(record);
+						} else {
+							loadAction = LoadAction.UPDATED;
+							entityManager.merge(record);
+						}
+
```

Comment:
That block is an indented mess, but that's not your fault. Can you just add a `// TODO: refactor this to be less of an indented mess` comment, which we can address later?